### DISCUSSION TITLE
ELI5: how are type=building relations used?

### DISCUSSION THREAD

**Discussion by ivanbranco** [StrandedKitty/streets-gl#59] [General]
Hi,

could someone explain to me what's the usage of type=building relations? When the renderer sees a type=building tag, what changes in its rendering?

**Comment by StrandedKitty** [comment]
Hi. If a `type=building` relation has a member with role `outline` (either way or relation) than this member doesn't get rendered.

**Comment by ivanbranco** [comment]
If a building:part is mapped, the outline isn't rendered even without type=building (and a type=building is created only when there's at least one building part). I don't understand the difference.

**Comment by StrandedKitty** [comment]
A way with `building:part=yes` exterminates all the `building=*` ways it intersects (with intersection area greater than some epsilon). `type=building` relations are meant to make this behaviour more predictable. If you have such a relation you can explicitly specify the outline and the parts and get predictable and very simple behavior: outline is hidden, parts are rendered.

Also, if a `building:part` is in a `type=building` relation and has a role `part` than it won't try to hide buildings intersecting with it. Because it is implied that the outline is hidden anyway, so there's no need for this.

**Comment by ivanbranco** [comment]
Ok, so it makes sense to suggest it mainly for "hanging" parts as the Wiki does, since they could intersect outlines of other buildings.

What happens tho if the "outline" member is tagged as building:part=*?

**Reply by StrandedKitty** [reply]
> What happens tho if the "outline" member is tagged as building:part=*?

It gets hidden anyway because it's the outline.
